File: LieInBed-App/src/main/java/io/github/ctimet/lieinbedapp/properties/PropertiesUtil.java
package io.github.ctimet.lieinbedapp.properties;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.HashMap;
import java.util.LinkedList;
import java.util.Properties;

public class PropertiesUtil {
    private static final Logger logger = LoggerFactory.getLogger(PropertiesUtil.class);

    public static void check(Properties pps, HashMap<String, ValueChecker<String>> checker, Properties defaultValue) {
        LinkedList<String> waitChange = new LinkedList<>();
        pps.forEach((k, v) -> {
            if (!checker.get((String) k).check((String) v)) {
                waitChange.add((String) k);
            }
        });
        waitChange.forEach(s -> {
            logger.warn("键 {} 对应的值 {} 无法通过 ValueChecker 检查，已恢复成默认值 {}",
                            s, pps.getProperty(s),            defaultValue.getProperty(s));
            pps.setProperty(s, defaultValue.getProperty(s));
        });
    }
}

File: LieInBed-App/src/main/kotlin/io/github/ctimet/lieinbedapp/properties/PropertiesUtil.kt
package io.github.ctimet.lieinbedapp.properties

import org.slf4j.LoggerFactory
import java.util.*

class PropertiesUtil {
    companion object {
        private val logger = LoggerFactory.getLogger(Properties::class.java)
        @JvmStatic
        fun check(check: Properties, checker: HashMap<String, ValueChecker<String>>, default: Properties) {
            val waitChange: LinkedList<Any> = LinkedList()
            check.forEach { k, v ->
                if (checker[k]?.check(v as String) == false) {
                    logger.info("键 $k 对应的值 $v 有问题，值 $v 无法通过ValueChecker检查，请重新设置。值已重置。")
                    waitChange.add(k)
                }
            }
            waitChange.forEach {
                check.remove(it)
            }
        }
    }
}